### PR TITLE
Break systemd service loop

### PR DESCRIPTION
**Pull request by ydirson**

### CODE REVIEW

File: SPECS/zfs.spec [hunk 1]
```diff
@@ -575,3 +576,11 @@ systemctl --system daemon-reload >/dev/null || true
 # ignore those files.
 %exclude /usr/share/initramfs-tools
 %endif
+
+%changelog
+* Mon May 06 2024 Yann Dirson <yann.dirson@vates.tech> 2.1.15-2
+- Remove dependency of zfs-import-cache.service against multipathd.service
+  to break a loop
+
+* Mon May 06 2024 Guillaume Thouvenin <guillaume.thouvenin@vates.tech> 2.1.15-1
```
Comment (stormi) [inline SPECS/zfs.spec]:
```suggestion
* Tue Feb 27 2024 Guillaume Thouvenin <guillaume.thouvenin@vates.tech> 2.1.15-1
```

File: SOURCES/zfs-2.1.15-break-multipathd-dependency-loop.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff -ru zfs-2.1.15.orig/etc/systemd/system/zfs-import-cache.service.in zfs-2.1.15/etc/systemd/system/zfs-import-cache.service.in
+--- zfs-2.1.15.orig/etc/systemd/system/zfs-import-cache.service.in	2024-02-27 00:04:26.084684081 +0100
++++ zfs-2.1.15/etc/systemd/system/zfs-import-cache.service.in	2024-05-03 16:15:44.434000000 +0200
+@@ -5,7 +5,8 @@
+ Requires=systemd-udev-settle.service
+ After=systemd-udev-settle.service
+ After=cryptsetup.target
+-After=multipathd.service
++#XCP-ng: break dependency loop, no ZFS over mpath?
```
Comment (gthvn1) [inline SOURCES/zfs-2.1.15-break-multipathd-dependency-loop.patch:9]: What is the loop dependency here? Are we sure that in any scenarios multipathd.service will be loaded after installing zfs? But in fact we don't need it currently as we don't use multipath... do we? So I think that it is ok.

Comment (ydirson) [reply SOURCES/zfs-2.1.15-break-multipathd-dependency-loop.patch:9]: We allow users to use it, as documented by XS:
* https://docs.xenserver.com/en-us/citrix-hypervisor/storage/multipath.html
* https://docs.xenserver.com/en-us/citrix-hypervisor/install/san-boot

I'll document the cycle better in the commit message.

**Review by gthvn1 — APPROVED** [review]

File: SOURCES/zfs-2.1.15-break-multipathd-dependency-loop.patch (hunk 2 above)
Comment (ydirson) [reply SOURCES/zfs-2.1.15-break-multipathd-dependency-loop.patch:9]: done

File: SPECS/zfs.spec (hunk 1 above)
Comment (ydirson) [reply SPECS/zfs.spec]: fixed

**Review by gthvn1 — APPROVED** [review]

File: SPECS/zfs.spec [hunk 3]
```diff
@@ -575,3 +576,11 @@ systemctl --system daemon-reload >/dev/null || true
 # ignore those files.
 %exclude /usr/share/initramfs-tools
 %endif
+
+%changelog
+* Mon May 06 2024 Yann Dirson <yann.dirson@vates.tech> 2.1.15-2
```
Comment (stormi) [inline SPECS/zfs.spec]: Since we are almost 2 months later, I think it would be good to update the date here.

**Review by tescande — APPROVED** [review]
Didn't find a better way to fix it, so approve this change...

File: SPECS/zfs.spec (hunk 3 above)
Comment (ydirson) [reply SPECS/zfs.spec]: done

**Review by stormi — APPROVED** [review]

**Comment by stormi** [comment]
Koji build to v8.3-incoming : http://koji.xcp-ng.org/taskinfo?taskID=62165

**Pull request merged**